hooks/checkout

```diff
@@ -39,8 +39,10 @@ if [ "${DRY_RUN}" != "true" ]; then
     git fetch -v --prune -- origin "refs/heads/${BUILDKITE_BRANCH}"
   fi
 
-  # Checkout the SHA
-  if [ "${BUILDKITE_COMMIT:-}" != "" ]; then
+  # Checkout the correct ref to continue the build
+  if [ "${BUILDKITE_COMMIT:-}" = "HEAD" || "${BUILDKITE_COMMIT}" = "" ]; then
```

Comment:
Whoops, when testing locally I realised this syntax is wrong, it should be:
```suggestion
  if [ "${BUILDKITE_COMMIT:-}" = "HEAD" -o "${BUILDKITE_COMMIT}" = "" ]; then
```

Reply:
Actually a better solution would be:
```suggestion
  if [ "${BUILDKITE_COMMIT:-}" = "HEAD" ] || [ "${BUILDKITE_COMMIT}" = "" ]; then
```

Reply:
I knew I was going to mess this up! Thank you!